src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java

```diff
@@ -1268,10 +1268,12 @@ public KeyDeserializer createKeyDeserializer(DeserializationContext ctxt,
         throws JsonMappingException
     {
         final DeserializationConfig config = ctxt.getConfig();
-        BeanDescription beanDesc = null;
-        KeyDeserializer deser = null;
-        if (_factoryConfig.hasKeyDeserializers()) {
-            beanDesc = config.introspectClassAnnotations(type);
+        final BeanDescription beanDesc = config.introspectClassAnnotations(type);
```

Comment:
With reference to the following code, I have determined that the process of trying `introspectClassAnnotations` again for `type.getRawClass()` is unnecessary, is this correct?
https://github.com/FasterXML/jackson-databind/blob/eeb5f2b764a505c91fda5c55620e60da30676152/src/main/java/com/fasterxml/jackson/databind/ser/std/BeanSerializerBase.java#L466

This may be a problem in cases where some customization has been done.

Reply:
Yes, getting raw class is pointless as far as I can see.